Add homing_currents for smooth TMC2130 homing

stahlfabrik opened the pull request.

Commit:
Added experimental support for homing currents

The Marlin Firmware unlike Klipper and Pruas Firmware has no concept of lowering currents for endstop-less TMC2130 homing. This commits adds homing currents setting to Configuration_adv.h and changes stepper currents before and after homing.

Untested, compiles.

stahlfabrik committed:
Improved #define handling thx to Motocoder

matthew-humphrey commented:
Am going to add this to Marlin/Configuration_adv.h (in disabled form), as it needs to be both places.

matthew-humphrey committed:
Add new SENSORELESS_HOMING_CURRENT feature to Marlin/Configuration_adv.h,
as this will be a requirement when we got to integrate with Marlin repo.
I also added a comment describing the feature.
